addons/website/models/website_visitor.py

```diff
@@ -250,28 +246,24 @@ def _create_visitor(self, website_track_values=None):
         return self.sudo().create(vals)
 
     def _cron_archive_visitors(self):
-        self.flush(['visit_count'])
-        yesterday = datetime.now() - timedelta(days=1)
-        query = """
-            UPDATE website_visitor
-            SET visit_count = visit_count + 1
-            WHERE id in (
-                SELECT visitor_id
-                FROM website_visitor_lastconnection
-                GROUP BY visitor_id
-                HAVING COUNT(*) > 1)
-        """
-        self.env.cr.execute(query, [yesterday])
-
-        query = """
-            DELETE FROM website_visitor_lastconnection
-            WHERE (visitor_id, connection_datetime) not in (
-               SELECT v.visitor_id, max(connection_datetime)
-               FROM website_visitor_lastconnection v
-               GROUP BY v.visitor_id)
-        """
-        self.env.cr.execute(query, [])
-
         one_week_ago = datetime.now() - timedelta(days=7)
         visitors_to_archive = self.env['website.visitor'].sudo().search([('last_connection_datetime', '<', one_week_ago)])
         visitors_to_archive.write({'active': False})
+
+    def _update_visitor_last_visit(self):
+        """ We need to do this part here to avoid concurrent updates error. """
+        with registry(self.env.cr.dbname).cursor() as cr:
+            date_now = datetime.now()
+            query = "UPDATE website_visitor SET"
+            if self.last_connection_datetime < (datetime.now() - timedelta(hours=8)):
+                query += "visit_count = visit_count + 1,"
```

Comment:
missing blank space : it will give `UPDATE website_visitor SETvisit_count = visit_count + 1,`